v-next/hardhat-keystore/src/utils.ts

```diff
@@ -0,0 +1,141 @@
+import type { Keystore } from "./types.js";
+
+import path from "node:path";
+
+import { HardhatPluginError } from "@ignored/hardhat-vnext-errors";
+import {
+  ensureDir,
+  exists,
+  readJsonFile,
+  writeJsonFile,
+} from "@ignored/hardhat-vnext-utils/fs";
+import chalk from "chalk";
+import envPaths from "env-paths";
+
+import { io } from "./io.js";
+import { PLUGIN_ID } from "./methods.js";
+import { setUpPassword } from "./password-manager.js";
+
+let keystoreCache: Keystore | undefined;
+let keystoreFilePath: string | undefined;
+
+// ATTENTION: For testing purposes
+export function setKeystoreCache(value: Keystore | undefined): void {
```

Comment:
Lets look at how we remove the need for this. I suspect if we passed keystore state in to these helper functions we would avoid the need for cache global.